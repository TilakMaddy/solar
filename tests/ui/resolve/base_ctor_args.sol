contract C {
    constructor(uint b) {}
}

contract D is C(3) {
    constructor() {}
}

contract E is C {
    constructor() C(5) {}
}

contract F is E { //~ERROR: contract makes multiple calls to base constructor C
    constructor() C(2) {} 
}

contract G is F {} //~ERROR: contract makes multiple calls to base constructor C
