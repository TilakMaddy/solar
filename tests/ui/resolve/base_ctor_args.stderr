error: contract makes multiple calls to base constructor C
  --> ROOT/tests/ui/resolve/base_ctor_args.sol:LL:CC
   |
LL | contract F is E {
   |          ^
   |

note: first call here
  --> ROOT/tests/ui/resolve/base_ctor_args.sol:LL:CC
   |
LL |     constructor() C(2) {} 
   |                   -
   |

note: second call here
  --> ROOT/tests/ui/resolve/base_ctor_args.sol:LL:CC
   |
LL |     constructor() C(5) {}
   |                   -
   |

error: contract makes multiple calls to base constructor C
  --> ROOT/tests/ui/resolve/base_ctor_args.sol:LL:CC
   |
LL | contract G is F {}
   |          ^
   |

note: first call here
  --> ROOT/tests/ui/resolve/base_ctor_args.sol:LL:CC
   |
LL |     constructor() C(2) {} 
   |                   -
   |

note: second call here
  --> ROOT/tests/ui/resolve/base_ctor_args.sol:LL:CC
   |
LL |     constructor() C(5) {}
   |                   -
   |

error: aborting due to 2 previous errors

